src/features/wallet/root/ActivityCard.tsx

```diff
@@ -1,102 +1,61 @@
-/* eslint-disable react/jsx-props-no-spreading */
-import React from 'react'
-import { Animated } from 'react-native'
-import SlidingUpPanel from 'rn-sliding-up-panel'
+import React, { useRef } from 'react'
 import { orderBy, random, times } from 'lodash'
-import CardHandle from './CardHandle'
-import Box from '../../../components/Box'
-import Text from '../../../components/Text'
+import BottomSheet from 'react-native-holy-sheet'
+import Animated from 'react-native-reanimated'
 import ActivityItem from './ActivityItem'
-import CarotDown from '../../../assets/images/carot-down.svg'
 import { WalletAnimationPoints } from './walletLayout'
-import { triggerNotification } from '../../../utils/haptic'
-import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import ActivityCardHeader from './ActivityCardHeader'
 
 type Props = {
-  animatedValue: Animated.Value
-  scrollOffset: Animated.Value
   animationPoints: WalletAnimationPoints
+  snapProgress?: Animated.SharedValue<number>
 }
 
-const ActivityCard = ({
-  animatedValue,
-  scrollOffset,
-  animationPoints,
-}: Props) => {
-  const renderItem = ({ item, index }: { item: TxnData; index: number }) => (
-    <ActivityItem
-      type={item.type}
-      time={item.time}
-      amount={item.amount}
-      isFirst={index === 0}
-      isLast={index === data.length - 1}
-    />
-  )
+type ActivityCardHandle = {
+  snapTo: (index?: number) => void
+}
 
-  const handlePress = () => {
-    triggerNotification()
-  }
+const ActivityCard = React.forwardRef(
```

Comment:
This is really cool stuff. One nitpick, you could destructure these `React.whatever`s in the import statement.
`import React, { useRef, Ref, useImperativeHandle, ElementRef, forwardRef } from 'react'`